src/modules/battery.rs

```diff
@@ -75,27 +75,76 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
 
 fn get_battery_status() -> Option<BatteryStatus> {
     let battery_manager = battery::Manager::new().ok()?;
-    match battery_manager.batteries().ok()?.next() {
-        Some(Ok(battery)) => {
-            log::debug!("Battery found: {:?}", battery);
-            let battery_status = BatteryStatus {
-                percentage: battery.state_of_charge().value * 100.0,
-                state: battery.state(),
-            };
+    let batteries = battery_manager.batteries().ok()?;
+    let battery_contructor = batteries
+        .filter_map(|battery| match battery {
+            Ok(battery) => {
+                log::debug!("Battery found: {:?}", battery);
+                Some(BatteryContructor {
+                    energy: battery.energy().value,
+                    energy_full: battery.energy_full().value,
+                    state: battery.state(),
+                })
+            }
+            Err(e) => {
+                log::debug!("Unable to access battery information:\n{}", &e);
+                None
+            }
+        })
+        .fold(
+            BatteryContructor {
+                energy: 0.0,
+                energy_full: 0.0,
+                state: battery::State::Unknown,
+            },
+            |mut acc, x| {
+                acc.energy += x.energy;
+                acc.energy_full += x.energy_full;
+                acc.state = battery_state_fold(acc.state, x.state);
+                acc
+            },
+        );
+    if battery_contructor.energy_full != 0.0 {
+        let battery = BatteryStatus {
+            percentage: battery_contructor.energy / battery_contructor.energy_full * 100.0,
```

Comment:
I can see this possibly returning a long decimal value. Perhaps we should limit the number of decimal places.

Reply:
Sorry, I do not understand, do you want to cast it to an integer value?

Reply:
My bad. I had forgotten that we round the value later on in the file:
https://github.com/starship/starship/blob/master/src/modules/battery.rs#L61